### PR TITLE
post-release: Changelog adjustments on `main` branch

### PR DESCRIPTION
**Pull request by kavirajk**
**What this PR does / why we need it**:
Adjust changelog on `main` branch post release of v2.9.0

Cleanup unreleased changes w.r.t changes went into v2.9.0

**Which issue(s) this PR fixes**:
Related: #10468 

**Special notes for your reviewer**:

**Checklist**
- [x] Reviewed the [`CONTRIBUTING.md`](https://github.com/grafana/loki/blob/main/CONTRIBUTING.md) guide (**required**)

- [x] `CHANGELOG.md` updated

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,16 +4,67 @@
 
 ##### Security
 
+#### Loki
+
+##### Enhancements
+* [10416](https://github.com/grafana/loki/pull/10416) **lpugoy**: Lambda-Promtail: Add support for WAF logs in S3
```
Comment (chaudum) [inline CHANGELOG.md:10]: Should this go underneath `Promtail`?

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,16 +4,67 @@
...
+* [10295](https://github.com/grafana/loki/pull/10295) **changhyuni**: Storage: remove signatureversionv2 from s3.
+* [10140](https://github.com/grafana/loki/pull/10140) **dannykopping**: Dynamic client-side throttling to avoid object storage rate-limits (GCS only)
+* [10302](https://github.com/grafana/loki/pull/10302) **ashwanthgoli**: Removes already deprecated `-querier.engine.timeout` CLI flag and corresponding YAML setting as well as the `querier.query_timeout` YAML setting.
+* [10301](https://github.com/grafana/loki/pull/10301) **wildum**: Promtail: users can now define `additional_fields` in cloudflare configuration.
```
Comment (chaudum) [inline CHANGELOG.md:14]: Should this go underneath `Promtail`?

File: CHANGELOG.md [hunk 3]
```diff
@@ -4,16 +4,67 @@
...
+* [10308](https://github.com/grafana/loki/pull/10308) **bboreham** Tracing: elide small traces for Stats call.
+* [10373](https://github.com/grafana/loki/pull/10373) **jeschkies**  Loki: Shard `avg_over_time` range aggregations.
+* [10377](https://github.com/grafana/loki/pull/10377) **shantanualsi** Remove deprecated config `-s3.sse-encryption` in favor or `-s3.sse.*` settings.
+* [10378](https://github.com/grafana/loki/pull/10378) **shantanualsi** Remove deprecated `ruler.wal-cleaer.period`
+* [10380](https://github.com/grafana/loki/pull/10380) **shantanualsi** Remove `experimental.ruler.enable-api` in favour of `ruler.enable-api`
+* [10395](https://github.com/grafana/loki/pull/10395/) **shantanualshi** Remove deprecated `split_queries_by_interval` and `forward_headers_list` configuration options in the `query_range` section
+* [10451](https://github.com/grafana/loki/pull/10451) **shantanualsi** Upgrade thanos `objstore`
```
Comment (chaudum) [inline CHANGELOG.md:21]: This line seems redundant to `Upgrade thanos objstore, dskit and other modules` (L30)

File: CHANGELOG.md [hunk 4]
```diff
@@ -4,16 +4,67 @@
...
+#### Loki
+
+##### Enhancements
+* [10416](https://github.com/grafana/loki/pull/10416) **lpugoy**: Lambda-Promtail: Add support for WAF logs in S3
+* [10295](https://github.com/grafana/loki/pull/10295) **changhyuni**: Storage: remove signatureversionv2 from s3.
+* [10140](https://github.com/grafana/loki/pull/10140) **dannykopping**: Dynamic client-side throttling to avoid object storage rate-limits (GCS only)
+* [10302](https://github.com/grafana/loki/pull/10302) **ashwanthgoli**: Removes already deprecated `-querier.engine.timeout` CLI flag and corresponding YAML setting as well as the `querier.query_timeout` YAML setting.
+* [10301](https://github.com/grafana/loki/pull/10301) **wildum**: Promtail: users can now define `additional_fields` in cloudflare configuration.
+* [10308](https://github.com/grafana/loki/pull/10308) **bboreham** Tracing: elide small traces for Stats call.
+* [10373](https://github.com/grafana/loki/pull/10373) **jeschkies**  Loki: Shard `avg_over_time` range aggregations.
```
Comment (chaudum) [inline CHANGELOG.md:16]:
```suggestion
* [10373](https://github.com/grafana/loki/pull/10373) **jeschkies** Shard `avg_over_time` range aggregations.
```

File: CHANGELOG.md (hunk 2 above)
Comment (chaudum) [reply CHANGELOG.md:14]: Then also the `Promtail: ` prefix to the line can be removed.

File: CHANGELOG.md (hunk 1 above)
Comment (kavirajk) [reply CHANGELOG.md:10]: agree. Just copied it from "unleased" as is. Will update it. 

Would be great if we can enforce this when adding to the changelog entry by the author in the first place.

File: CHANGELOG.md (hunk 3 above)
Comment (kavirajk) [reply CHANGELOG.md:21]: Both seems to be different PRs. Looked into the PRs and both seems to upgrading thanos packages/modules. Going to leave it like this. @shantanualsi can you fix these entries if you think it's irrelevant.

File: CHANGELOG.md (hunk 2 above)
Comment (kavirajk) [reply CHANGELOG.md:14]: Done 👍

File: CHANGELOG.md (hunk 1 above)
Comment (kavirajk) [reply CHANGELOG.md:10]: Done 👍

File: CHANGELOG.md (hunk 3 above)
Comment (shantanualsi) [reply CHANGELOG.md:21]: objstore was upgraded once again later to fix another issue..  should I merge the. changelog entry?

**Pull request merged**